### ISSUE TITLE
Use a more common template system

### ISSUE THREAD

**Issue by theSage21** [PyJaipur/PyJudge#124]
While bottle provides a simple template system, almost everyone coming to this project has been already exposed to some other template system. This causes the misuse of bottle's own template system. [Using import instead of rebase](https://github.com/PyJaipur/PyJudge/blob/e3debe1e15447a6faac9d509e7ad5283731fa2c0/views/error.html#L1) for example is a common mistake.

We can instead use the more common Jinja2 template system. Relevant documentation can be found in [bottle's documentation](https://bottlepy.org/docs/dev/api.html#bottle.template)

**Comment by rohitjain00** [comment]
Yes or I think we should move the entire front end on a front end based framework like angular Js and use server as an API for sending and receiving data? I think this is more efficient tan retrieving whole web pages from the server.

**Comment by theSage21** [comment]
It's not better than the existing system in any way so far. Both approaches are equivalent. If you would like to take that approach I suggest using [svelte](https://svelte.dev/) as the UI framework.

**Comment by rohitjain00** [comment]
Is there any specific reason to use Svelte?

**Comment by theSage21** [comment]
- it's smaller
- it's cleaner while writing code
- it's not popular so you have to actually learn it to use it instead of copy pasting from some blog :smile:

**Comment by theSage21** [comment]
I'd recommend going through <https://www.youtube.com/watch?v=AdNJ3fydeao> to understand how svelte might be superior to commonplace things like React/Vue/Angular. To see a simple benchmark I'd recommend [this one](https://www.freecodecamp.org/news/a-realworld-comparison-of-front-end-frameworks-with-benchmarks-2019-update-4be0d3c78075/)
